build/common.sh

```diff
@@ -730,8 +730,6 @@ function kube::build::sync_to_container() {
     --filter='- /_tmp/' \
     --filter='- /_output/' \
     --filter='- /' \
-    --filter='H zz_generated.*' \
```

Comment:
Correct, part of the intent is to combine the protobuf script into the codegen script. 

The reason for this is because this PR adds generated protobuf to a subpackage (sample apiserver). Sub-packages have their own hack/update-codegen which calls the code-generator package facilities. These sub-package scripts do not support protobuf generation.

Instead of adding a new protobuf generation script for each sub-package, I combined the two scripts into a single script in kubernetes/hack/, and mirror this new protobuf generation in the sub-package update-codegen scripts so that they're uniform.


The dockerization part as far as I can tell is only because protoc doesn't appear to be available on the bare CI machines, and the build docker images are configured to have protoc. I'm pretty sure if protoc could be on the bare CI machines then this script could remain un-dockerized.


Further, I had added the line above because I had to make this update-codegen script un-dockerized. What I had found was that verify-codegen.sh would fail no matter what, because the rsync config for the docker build image has the zz_generated.* files removed. Thus when we go to verify newly generated codegen against existing, the zz_generated.* files aren't on the docker image, so everything looks like it never existed before.


The above made sense to me at the time, however I have since rebuilt and re-added the line above and am not able to reproduce the issue. I'm going to continue testing for a bit and will update when I have an update.

Reply:
Ah, I looked deeper into the issue and it makes more sense to me now.

I hadn't seen until now that some generated files are actually created via a Make rule.

Specifically, I had noticed that zz_generated.deepcopy files appeared to be updated by running verify-codegen.sh if they were out of date. With the change above, this would be reported by running verify-codegen.sh that you need to run update-codegen.sh to resolve.

This is not correct, as I've now found, since those generated files aren't handled by this system.

At least, this is my current understanding. I will reverse the above change.

Reply:
I have amended my previous commit to reverse the above change.

I also amended to include changes from commit ff46c0f which affect hack/update-codegen.sh and hack/verify-codegen.sh and needed to be reconciled with my changes.